### DISCUSSION TITLE
First Time Conda Install - "Always ok" test failing

### DISCUSSION THREAD

**Discussion by crswong888** [idaholab/moose#16576] [Q&A Getting Started] [answered]
Hello, I've never used the Conda MOOSE environment before. Surprisingly, I got all the way up to compiling the test app with no issues! Sadly, TestHarness is doing some weird things. For example:

```
(moose) inl602177:~ christopherwong$ cd projects/moose/
(moose) inl602177:moose christopherwong$ cd test/
(moose) inl602177:test christopherwong$ make -j4
(moose) inl602177:test christopherwong$ ./run_tests -i always_ok -p 2
tests/test_harness.always_ok: Working Directory: /Users/christopherwong/projects/moose/test/tests/test_harness
tests/test_harness.always_ok: Running command: mpiexec -n 2 /Users/christopherwong/projects/moose/test/moose_test-opt -i good.i --error --error-unused --error-override --no-gdb-backtrace
tests/test_harness.always_ok: Fatal error in MPI_Init_thread: Invalid group, error stack:
tests/test_harness.always_ok: MPIR_Init_thread(586)..............: 
tests/test_harness.always_ok: MPID_Init(224).....................: channel initialization failed
tests/test_harness.always_ok: MPIDI_CH3_Init(105)................: 
tests/test_harness.always_ok: MPID_nem_init(324).................: 
tests/test_harness.always_ok: MPID_nem_tcp_init(175).............: 
tests/test_harness.always_ok: MPID_nem_tcp_get_business_card(401): 
tests/test_harness.always_ok: MPID_nem_tcp_init(373).............: gethostbyname failed, inl602177 (errno 0)
tests/test_harness.always_ok: (unknown)(): Invalid group
tests/test_harness.always_ok: 
tests/test_harness.always_ok: 
tests/test_harness.always_ok: Exit Code: 8
tests/test_harness.always_ok: ################################################################################
tests/test_harness.always_ok: Tester failed, reason: CRASH
tests/test_harness.always_ok: 
tests/test_harness.always_ok ...................................................... [OVERSIZED] FAILED (CRASH)


Final Test Results:
--------------------------------------------------------------------------------------------------------------
tests/test_harness.always_ok ...................................................... [OVERSIZED] FAILED (CRASH)
--------------------------------------------------------------------------------------------------------------
Ran 1 tests in 1.3 seconds.
0 passed, 0 skipped, 0 pending, 1 FAILED
(moose) inl602177:test christopherwong$ 
```

Anyone got a clue what's going on?

**Comment by cticenhour** [comment, marked as answer]
You need to add your hostname to your `/etc/hosts` file referencing`127.0.0.1` - your localhost address.

First run the command `hostname` in Terminal.

Let's say that returns `mycoolname`. Then using `sudo`, edit `/etc/hosts` to add `127.0.0.1    mycoolname` to the last line.

A sample `hosts` file and a bit more information about the gethostbyname issue can be found on the MOOSE website [here](https://mooseframework.inl.gov/help/troubleshooting.html#failingtests)

**Reply by cticenhour** [reply]
I also see that you're on an INL machine - the error number 0 (as opposed to 3 in the linked FAQ) popped up recently on a new hire's Mac as well. If the following tip doesn't work and you are on a Mac, also try to turn off the Firewall (look in Security & Privacy in System Preferences). If that doesn't work, and Jamf management software is installed, you might need to get special permission to change some configuration settings.

**Reply by cticenhour** [reply]
Tagging @aeslaughter

**Reply by crswong888** [reply]
okay your answer here was 100% the solution to the "always ok" test failing. I also had another issue where my mac kept asking permission to run the test binary - for every single regression test - and what we talked about on slack solved that too. Seems like I got everything in working order now :)
